contracts/WooCrossChainRouterV2.sol

```diff
@@ -172,6 +150,40 @@ contract WooCrossChainRouterV2 is IWooCrossChainRouterV2, Ownable, ReentrancyGua
         );
     }
 
+    function onOFTReceived(
+        uint16 srcChainId,
+        bytes memory, // srcAddress
+        uint64, // nonce
+        bytes32 from,
+        uint256 amountLD,
+        bytes memory payload
+    ) external {
+        address oft = _msgSender();
+        require(allowBridgeOFTs[sgChainIdLocal][oft], "WooCrossChainRouterV2: INVALID_CALLER");
+        require(
+            wooCrossChainRouters[srcChainId] == address(uint160(uint256(from))),
+            "WooCrossChainRouterV2: INVALID_FROM"
+        );
+        // msgSender should be OFT address if requires above are passed
+
+        // make sure the same order to _encodePayload() when decode payload
+        (uint256 refId, address to, address toToken, uint256 minToAmount) = abi.decode(
+            payload,
+            (uint256, address, address, uint256)
+        );
+
+        address bridgedToken;
+        if (isProxyOFTs[sgChainIdLocal][oft]) {
+            bridgedToken = IOFTWithFee(oft).token();
+        } else {
+            bridgedToken = oft;
+        }
```

Comment:
In fact, we always use oft.token() should be good to go, right?

Reply:
yes, fix already